File: backend/src/main/java/com/example/fiscos/dto/invoice/CompleteInvoiceDTO.java
package com.example.fiscos.dto.invoice;

import java.util.List;

import com.example.fiscos.dto.product.ProductDTO;
import com.example.fiscos.dto.qrCode.QrCodeDTO;
import com.example.fiscos.dto.supplier.SupplierDTO;
import com.example.fiscos.dto.tax.TaxesDTO;
import com.example.fiscos.dto.user.UserDTO;
import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class CompleteInvoiceDTO extends InvoiceDTO {

    @JsonProperty("user")
    private UserDTO user;

    @JsonProperty("supplier")
    private SupplierDTO supplier;

    @JsonProperty("taxes")
    private TaxesDTO taxes;

    @JsonProperty("products")
    private List<ProductDTO> products;

    @JsonProperty("qrcode")
    private QrCodeDTO qrcode;
}

File: frontend/dto/CompleteInvoiceDTO.ts
import { ProductDTO } from "./ProductDTO";
import { QrCodeDTO } from "./QrCodeDTO";
import { SupplierDTO } from "./SupplierDTO";
import { TaxesDTO } from "./TaxesDTO";
import { UserDTO } from "./UserDTO";

export interface CompleteInvoiceDTO {
  id: number;
  issueDate: number;
  totalValue: number;
  user: UserDTO;
  supplier: SupplierDTO;
  taxes: TaxesDTO;
  products: ProductDTO[];
  qrcode: QrCodeDTO;
}
